src/bundle.c

```diff
@@ -171,30 +175,242 @@ static int unload_tracked_bundle(const char *bundle_name, struct list **subs)
 	return EBUNDLE_NOT_TRACKED;
 }
 
-/* Check if bundle is included by any subscribed bundles */
-static bool is_included(const char *bundle_name, struct manifest *mom)
+/* Return list of bundles that include bundle_name */
+void required_by(struct list **reqd_by, const char *bundle_name, struct manifest *mom, int rl)
 {
-	bool ret = false;
 	struct list *b, *i;
+	// track recursion level for indentation
+	rl++;
 
 	b = list_head(mom->submanifests);
 	while (b) {
 		struct manifest *bundle = b->data;
 		b = b->next;
 
+		int indent = 0;
 		i = list_head(bundle->includes);
 		while (i) {
 			char *name = i->data;
 			i = i->next;
 
 			if (strcmp(name, bundle_name) == 0) {
-				ret = true;
-				goto out;
+				char *bundle_str = NULL;
+				indent = (rl - 1) * 4;
+				if (rl == 1) {
+					string_or_die(&bundle_str, "%*s* %s\n", indent + 2, "", bundle->component);
+				} else {
+					string_or_die(&bundle_str, "%*s|-- %s\n", indent, "", bundle->component);
+				}
+
+				*reqd_by = list_append_data(*reqd_by, bundle_str);
+				required_by(reqd_by, bundle->component, mom, rl);
 			}
 		}
 	}
+}
+/* Return recursive list of included bundles */
+int show_included_bundles(char *bundle_name)
```

Comment:
For now, I can move the swupd_init/deinit out of the two options introduced here.